pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotAggregateExchangeNodeInsertRule.java

```diff
@@ -119,23 +132,25 @@ public void onMatch(RelOptRuleCall call) {
     }
 
     // 3. attach intermediate agg stage.
-    RelNode newAggNode = makeNewIntermediateAgg(call, oldAggRel, exchange);
+    RelNode newAggNode = makeNewIntermediateAgg(call, oldAggRel, exchange, true, null, null);
     call.transformTo(newAggNode);
   }
 
-  private RelNode makeNewIntermediateAgg(RelOptRuleCall ruleCall, Aggregate oldAggRel, LogicalExchange exchange) {
+  /**
+   * Creates an intermediate stage aggregation. If argList and groupByList are passed by the caller, they are used to
+   * create the new aggregation function. If not, they are fetched from the existing aggregation function with the
+   * assumption that the intermediate aggregation is done after a leaf aggregation and exchange.
+   */
+  private RelNode makeNewIntermediateAgg(RelOptRuleCall ruleCall, Aggregate oldAggRel, LogicalExchange exchange,
+      boolean isLeafStageAggregationPresent, List<Integer> argList, List<Integer> groupByList) {
 
     // add the exchange as the input node to the relation builder.
     RelBuilder relBuilder = ruleCall.builder();
     relBuilder.push(exchange);
-    List<RexNode> inputExprs = new ArrayList<>(relBuilder.fields());
 
     // make input ref to the exchange after the leaf aggregate.
     RexBuilder rexBuilder = exchange.getCluster().getRexBuilder();
     final int nGroups = oldAggRel.getGroupCount();
-    for (int i = 0; i < nGroups; i++) {
```

Comment:
why was this removed? this sets up the inputrefs for the exchange, right?

Reply:
This seemed unnecessary based on my code reading. But I am not sure, so adding it back.